### ISSUE TITLE
Cursor disappears frequently, when only one or no apps are running.

### ISSUE THREAD

**Issue by Sinaaaa** [hyprwm/Hyprland#4269]
### Hyprland Version

v0.33.1-97-gf86cdcf8

### Bug or Regression?

Bug

### Description
(`env = WLR_NO_HARDWARE_CURSORS,1` is set)
The cursor likes disappearing when the starting point is either an empty workspace with no apps running or I'm closing the app running with my keybind. Violently moving the mouse around makes the cursor reappear, dragging it to the left to my vertical Waybar does it too. However the issue happens even if Waybar is killed. 

If I have a very tiny floating window open -dragged far out of the way, but still within the active workspace- the issue does not occur.

I tried various versions of Hyprland. Latest git from AUR, a week old git from a snapshot & the normal Arch package, they all seem to produce this issue.
(I run Hyprland from an intel IGPU & all my displays are connected to this IGPU, while I have an AMD dedicated GPU that I only use for gaming. Disabling all but one of my displays does not fix it)

### How to reproduce

For example use a keybind to run` rofi -show window` & then exit from rofi with 'esc'.  

Alternatively I can just start up `alacritty` & close it with my keybind. 

### Crash reports, logs, images, videos

https://github.com/hyprwm/Hyprland/assets/8416043/a553c69b-533d-403a-842c-7897ad03594c

**Comment by Sinaaaa** [comment]
Seems like this might be a wlroots issue, because the same thing also happens in Labwc, however there the tiniest movement of the mouse is enough to recover the cursor, which cannot be said about Hyprland.

**Comment by vaxerski** [comment]
it's not a wlroots issue. It was working a while back but I broke it some time ago.

**Comment by Sinaaaa** [comment]
That's good to know, at least I won't fret over trying to fix something that I thought could be due to the niche-ness of my system.

**Comment by Sinaaaa** [comment]
Since yesterday the behavior has changed. (improved a little)

The cursor does not disappear anymore when closing standalone apps on a workspace like Alacritty. However the cursor is stuck on the text indicator instead. (or any other indicator the app requested before closing it) 
The cursor does still disappear if I close either of the rofi windows with esc.

**Comment by Sinaaaa** [comment]
I have not compiled Hyprland for a few days & after updating just now this issue seems to be solved. (I'll reopen it later, if it's a fluke)

**Comment by nonetrix** [comment]
Still happening for me on AMD GPU on NixOS

**Comment by Sinaaaa** [comment]
Yeah, I will open it again. What I perceived as fixed was merely Rofi-lbonn changing cursor behavior on their side, but it still happens.

**Comment by ligmaSec** [comment]
> Still happening for me on AMD GPU on NixOS, my workaround is to just hover over Waybar. Pretty sure I am running newest version as I am using the Hyprland flake

I second this, for some reason my cursor disappears too when I open mpv

**Comment by atikiNBTW** [comment]
> Still happening for me on AMD GPU on NixOS, my workaround is to just hover over Waybar. Pretty sure I am running newest version as I am using the Hyprland flake
Same, but this started happen only after I changed the default cursor

**Comment by nonetrix** [comment]
I wonder if Hyprcursor bypasses it to some extent lol haven't tried, I am still having this as well though but I'll admit I'm on slightly out of date version of Hyprland now
